test/jdk/java/awt/print/PrinterJob/PrintNullString.java

```diff
@@ -21,79 +21,43 @@
+import java.lang.Override;
 import java.text.AttributedCharacterIterator;
 import java.text.AttributedString;
 
-import javax.swing.JOptionPane;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+import javax.print.attribute.standard.Destination;
 
 /*
  * @test
  * @bug 4223328
- * @summary Printer graphics must behave the same as screen graphics
+ * @summary Printer graphics must throw expected exceptions
  * @key printer
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
- * @run main/manual PrintNullString
+ * @run main PrintNullString
  */
 public class PrintNullString extends Frame {
-    private static final String INSTRUCTIONS =
-            "This test should print a page which contains the same\n" +
-            "text messages as in the test window on the screen.\n" +
-            "\n" +
-            "The messages should contain only 'OK' and 'expected' messages.\n" +
-            "Press Pass if it's the case; otherwise press Fail.\n" +
-            "\n" +
-            "If the page fails to print, but there were no exceptions\n" +
-            "then the problem is likely elsewhere (i.e. your printer)";
 
     public static void main(String[] args) throws Exception {
         if (PrinterJob.lookupPrintServices().length == 0) {
             throw new RuntimeException("Printer not configured or available.");
         }
-
-        PassFailJFrame.builder()
-                .instructions(INSTRUCTIONS)
-                .testUI(PrintNullString::new)
-                .rows((int) INSTRUCTIONS.lines().count() + 1)
-                .columns(45)
-                .build()
-                .awaitAndCheck();
-    }
-
-    public PrintNullString() {
-        super("PrintNullString");
-
-        TextCanvas c = new TextCanvas();
-        add("Center", c);
-
-        Button b = new Button("Print");
-        add("South", b);
-        b.addActionListener(e -> {
-            PrinterJob pj = PrinterJob.getPrinterJob();
-            if (pj.printDialog()) {
-                pj.setPrintable(c);
-                try {
-                    pj.print();
-                } catch (PrinterException ex) {
-                    ex.printStackTrace();
-                    String msg = "PrinterException: " + ex.getMessage();
-                    JOptionPane.showMessageDialog(b, msg, "Error occurred",
-                            JOptionPane.ERROR_MESSAGE);
-                    PassFailJFrame.forceFail(msg);
-                }
-            }
-        });
-        pack();
+        PrinterJob pj = PrinterJob.getPrinterJob();
+        pj.setPrintable(new TextCanvas(), new PageFormat());
+        PrintRequestAttributeSet pSet = new HashPrintRequestAttributeSet();
+        File file = new File("out.prn");
+        file.deleteOnExit();
+        pSet.add(new Destination(file.toURI()));
+        pj.print(pSet);
     }
 
     private static class TextCanvas extends Panel implements Printable {
```

Comment:
```suggestion
    private static class PrintText implements Printable {
```
`Panel` isn't needed any more, so simple class would do.

Better yet, make the test class `PrintNullString` implement `Printable` and embed everything from `TextCanvas` directly to `PrintNullString`.